src/test/resources/countNonEmptyForm.xml

```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
This form feels overcomplicated to test the behavior of `count-non-empty` and `count` expressions. Would you consider simplifying it so that future readers can get a quicker sense of how it relates to the assertions in the test?

Reply:
Ok, I simplified it.